Panic on Policy Add Request

Coming from #14052

## Bug report

On k8s 1.14.10
Cilium v1.8.4

**General Information**

- Cilium version 1.8.4
- Kernel version 4.9
- Kubernetes 1.14.10
- Installed with Kops

**How to reproduce the issue**

Having:

A namespace with an application la jupyterHub installed with Helm
```
`helm upgrade -i jupyterhub jupyterhub/jupyterhub --namespace recommender -f values.yaml --version=0.10.2`
```
I do an upgrade to replace the node this app is living.

Upgrade will fail as cilium will be unable to start giving the following errors:
```
level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"singleuser-server\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}} NodeSelector:{} Ingress:[{FromEndpoints:[{\"matchLabels\":{\"k8s:hub.jupyter.org/network-access-singleuser\":\"true\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\"}}] FromRequires:[] ToPorts:[{Ports:[{Port:notebook-port Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]}] Egress:[{ToEndpoints:[{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"hub\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}}] ToRequires:[] ToPorts:[{Ports:[{Port:8081 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[{}] ToRequires:[] ToPorts:[{Ports:[{Port:53 Protocol:UDP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>} {Ports:[{Port:53 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[] ToRequires:[] ToPorts:[] ToCIDR: ToCIDRSet:[0.0.0.0/0-[169.254.169.254/32]] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[{LabelSelector:0xc000ee3c40 requirements:0xc000ee3cc0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.0.0.0.0/1: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ee3d20 requirements:0xc000ee3da0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.192.0.0.0/2: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ee3de0 requirements:0xc000ee3e60 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.128.0.0.0/3: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ee3ea0 requirements:0xc000ee3f20 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.176.0.0.0/4: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ee3f60 requirements:0xc000ee3fe0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.160.0.0.0/5: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe020 requirements:0xc000efe0a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.172.0.0.0/6: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe0e0 requirements:0xc000efe160 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.170.0.0.0/7: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe1a0 requirements:0xc000efe220 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.168.0.0.0/8: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe260 requirements:0xc000efe2e0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.0.0.0/9: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe320 requirements:0xc000efe3a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.128.0.0/10: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe3e0 requirements:0xc000efe460 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.192.0.0/11: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe4a0 requirements:0xc000efe520 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.224.0.0/12: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe560 requirements:0xc000efe5e0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.240.0.0/13: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe620 requirements:0xc000efe6a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.248.0.0/14: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe6e0 requirements:0xc000efe760 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.252.0.0/15: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe7a0 requirements:0xc000efe820 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.255.0.0/16: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe860 requirements:0xc000efe8e0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.0.0/17: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe920 requirements:0xc000efe9a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.192.0/18: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efe9e0 requirements:0xc000efea60 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.128.0/19: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efeaa0 requirements:0xc000efeb20 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.176.0/20: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efeb60 requirements:0xc000efebe0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.160.0/21: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efec20 requirements:0xc000efeca0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.172.0/22: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efece0 requirements:0xc000efed60 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.170.0/23: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efeda0 requirements:0xc000efee20 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.168.0/24: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efee60 requirements:0xc000efeee0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.0/25: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efef20 requirements:0xc000efefa0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.128/26: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000efefe0 requirements:0xc000eff060 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.192/27: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000eff0a0 requirements:0xc000eff120 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.224/28: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000eff160 requirements:0xc000eff1e0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.240/29: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000eff220 requirements:0xc000eff2a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.248/30: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000eff2e0 requirements:0xc000eff360 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.252/31: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000eff3a0 requirements:0xc000eff420 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.255/32: ,},MatchExpressions:[]LabelSelectorRequirement{},}}]}] Labels:[k8s:io.cilium.k8s.policy.derived-from=NetworkPolicy k8s:io.cilium.k8s.policy.name=singleuser k8s:io.cilium.k8s.policy.namespace=recommender k8s:io.cilium.k8s.policy.uid=e82adfaf-34ab-11eb-9d7a-0296c41eb8a2] Description:}]" policyAddRequest=ca797604-34b0-11eb-a2d8-0a92039cbfa8 subsys=daemon
panic: runtime error: index out of range [3] with length 0

goroutine 145 [running]:
encoding/binary.bigEndian.Uint32(...)
	/usr/local/go/src/encoding/binary/binary.go:112
github.com/cilium/cilium/pkg/byteorder.HostSliceToNetwork(0x0, 0x0, 0x0, 0xa, 0xc000f995f8, 0xc000f9aeb4)
	/go/src/github.com/cilium/cilium/pkg/byteorder/byteorder.go:134 +0x1ee
github.com/cilium/cilium/pkg/datapath/linux/config.(*HeaderfileWriter).WriteNodeConfig(0x4116998, 0x27fb660, 0xc000f8e6a0, 0xc0000c3a48, 0xc000cc25b0, 0xc000f8e690)
	/go/src/github.com/cilium/cilium/pkg/datapath/linux/config/config.go:108 +0x5baf
github.com/cilium/cilium/pkg/datapath/loader.hashDatapath(0x7f35782a6338, 0xc0002a0a80, 0xc0000c3a48, 0x0, 0x0, 0x0, 0x0, 0x40cfd8)
	/go/src/github.com/cilium/cilium/pkg/datapath/loader/hash.go:56 +0x109
github.com/cilium/cilium/pkg/datapath/loader.(*objectCache).Update(0xc0002165a0, 0xc0000c3a48)
	/go/src/github.com/cilium/cilium/pkg/datapath/loader/cache.go:139 +0x60
github.com/cilium/cilium/pkg/datapath/loader.newObjectCache(0x7f35782a6338, 0xc0002a0a80, 0xc0000c3a48, 0xc000050380, 0x15, 0xc00003a000)
	/go/src/github.com/cilium/cilium/pkg/datapath/loader/cache.go:120 +0x135
github.com/cilium/cilium/pkg/datapath/loader.NewObjectCache(...)
	/go/src/github.com/cilium/cilium/pkg/datapath/loader/cache.go:133
github.com/cilium/cilium/pkg/datapath/loader.(*Loader).init.func1()
	/go/src/github.com/cilium/cilium/pkg/datapath/loader/loader.go:84 +0x68
sync.(*Once).doSlow(0xc00051dcc0, 0xc000ecb470)
	/usr/local/go/src/sync/once.go:66 +0xec
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:57
github.com/cilium/cilium/pkg/datapath/loader.(*Loader).init(0xc00051dcc0, 0x7f35782a6338, 0xc0002a0a80, 0xc0000c3a48)
	/go/src/github.com/cilium/cilium/pkg/datapath/loader/loader.go:83 +0xa2
github.com/cilium/cilium/pkg/datapath/loader.(*Loader).Reinitialize(0xc00051dcc0, 0x285d500, 0xc00062a000, 0x28710c0, 0xc000338380, 0x2329, 0x7f35782a62e0, 0xc0002a0a80, 0x27fbc60, 0xc000217e00, ...)
	/go/src/github.com/cilium/cilium/pkg/datapath/loader/base.go:187 +0x2c0
github.com/cilium/cilium/daemon/cmd.(*Daemon).policyAdd(0xc000338380, 0xc0001cc170, 0x1, 0x1, 0xc000efde80, 0xc000619aa0)
	/go/src/github.com/cilium/cilium/daemon/cmd/policy.go:285 +0x1ff7
github.com/cilium/cilium/daemon/cmd.(*PolicyAddEvent).Handle(0xc000a95b90, 0xc000619aa0)
	/go/src/github.com/cilium/cilium/daemon/cmd/policy.go:217 +0x5c
github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).run.func1()
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:256 +0x16b
sync.(*Once).doSlow(0xc00048f518, 0xc000a07fb8)
	/usr/local/go/src/sync/once.go:66 +0xec
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:57
github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).run(0xc00048f500)
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:244 +0x65
created by github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).Run
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:240 +0x62
```

If I restart the agents, new ones will be failing too but running ones and agents on master will still be working.

Current solution, delete node when only one, and manually terminate the instance. New one came and it handle the Policy ok:
```
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"hub\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}} NodeSelector:{} Ingress:[{FromEndpoints:[{\"matchLabels\":{\"k8s:hub.jupyter.org/network-access-hub\":\"true\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\"}}] FromRequires:[] ToPorts:[{Ports:[{Port:http Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]}] Egress:[{ToEndpoints:[{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"proxy\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}}] ToRequires:[] ToPorts:[{Ports:[{Port:8001 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"singleuser-server\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}}] ToRequires:[] ToPorts:[{Ports:[{Port:8888 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[{}] ToRequires:[] ToPorts:[{Ports:[{Port:53 Protocol:UDP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>} {Ports:[{Port:53 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[] ToRequires:[] ToPorts:[] ToCIDR: ToCIDRSet:[0.0.0.0/0] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[{LabelSelector:0xc0003deb40 requirements:0xc0003debe0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{reserved.world: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000f0dc60 requirements:0xc000f0dd20 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.0.0.0.0/0: ,},MatchExpressions:[]LabelSelectorRequirement{},}}]}] Labels:[k8s:io.cilium.k8s.policy.derived-from=NetworkPolicy k8s:io.cilium.k8s.policy.name=hub k8s:io.cilium.k8s.policy.namespace=recommender k8s:io.cilium.k8s.policy.uid=e80ba7fe-34ab-11eb-9d7a-0296c41eb8a2] Description:}]" policyAddRequest=e80c947d-34ab-11eb-a8dc-067e3082833c subsys=daemon
level=info msg="Policy imported via API, recalculating..." policyAddRequest=e80c947d-34ab-11eb-a8dc-067e3082833c policyRevision=2 subsys=daemon
level=info msg="NetworkPolicy successfully added" k8sApiVersion= k8sNetworkPolicyName=hub subsys=k8s-watcher
level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"proxy\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}} NodeSelector:{} Ingress:[{FromEndpoints:[{}] FromRequires:[] ToPorts:[{Ports:[{Port:http Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>} {Ports:[{Port:https Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]} {FromEndpoints:[{\"matchLabels\":{\"k8s:hub.jupyter.org/network-access-proxy-http\":\"true\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\"}}] FromRequires:[] ToPorts:[{Ports:[{Port:http Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]} {FromEndpoints:[{\"matchLabels\":{\"k8s:hub.jupyter.org/network-access-proxy-api\":\"true\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\"}}] FromRequires:[] ToPorts:[{Ports:[{Port:api Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]}] Egress:[{ToEndpoints:[{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"hub\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}}] ToRequires:[] ToPorts:[{Ports:[{Port:8081 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"singleuser-server\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}}] ToRequires:[] ToPorts:[{Ports:[{Port:8888 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[{}] ToRequires:[] ToPorts:[{Ports:[{Port:53 Protocol:UDP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>} {Ports:[{Port:53 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[] ToRequires:[] ToPorts:[] ToCIDR: ToCIDRSet:[0.0.0.0/0] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[{LabelSelector:0xc0003deb40 requirements:0xc0003debe0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{reserved.world: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc00103d620 requirements:0xc00103d7c0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.0.0.0.0/0: ,},MatchExpressions:[]LabelSelectorRequirement{},}}]}] Labels:[k8s:io.cilium.k8s.policy.derived-from=NetworkPolicy k8s:io.cilium.k8s.policy.name=proxy k8s:io.cilium.k8s.policy.namespace=recommender k8s:io.cilium.k8s.policy.uid=e82adef0-34ab-11eb-9d7a-0296c41eb8a2] Description:}]" policyAddRequest=e82bb987-34ab-11eb-a8dc-067e3082833c subsys=daemon
level=info msg="NetworkPolicy successfully added" k8sApiVersion= k8sNetworkPolicyName=proxy subsys=k8s-watcher
level=info msg="Policy imported via API, recalculating..." policyAddRequest=e82bb987-34ab-11eb-a8dc-067e3082833c policyRevision=3 subsys=daemon
level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"singleuser-server\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}} NodeSelector:{} Ingress:[{FromEndpoints:[{\"matchLabels\":{\"k8s:hub.jupyter.org/network-access-singleuser\":\"true\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\"}}] FromRequires:[] ToPorts:[{Ports:[{Port:notebook-port Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]}] Egress:[{ToEndpoints:[{\"matchLabels\":{\"k8s:app\":\"jupyterhub\",\"k8s:component\":\"hub\",\"k8s:io.kubernetes.pod.namespace\":\"recommender\",\"k8s:release\":\"jupyterhub\"}}] ToRequires:[] ToPorts:[{Ports:[{Port:8081 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[{}] ToRequires:[] ToPorts:[{Ports:[{Port:53 Protocol:UDP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>} {Ports:[{Port:53 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ToCIDR: ToCIDRSet:[] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[]} {ToEndpoints:[] ToRequires:[] ToPorts:[] ToCIDR: ToCIDRSet:[0.0.0.0/0-[169.254.169.254/32]] ToEntities:[] ToServices:[] ToFQDNs:[] ToGroups:[] aggregatedSelectors:[{LabelSelector:0xc000ecf660 requirements:0xc000ecf760 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.0.0.0.0/1: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ecf7e0 requirements:0xc000ecf8a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.192.0.0.0/2: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ecf900 requirements:0xc000ecfbe0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.128.0.0.0/3: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ecfc20 requirements:0xc000ecfca0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.176.0.0.0/4: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ecfce0 requirements:0xc000ecfd60 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.160.0.0.0/5: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ecfda0 requirements:0xc000ecfe40 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.172.0.0.0/6: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ecfe80 requirements:0xc000ecff00 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.170.0.0.0/7: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ecff60 requirements:0xc000ecffe0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.168.0.0.0/8: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddc0e0 requirements:0xc000ddc1c0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.0.0.0/9: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddc240 requirements:0xc000ddc3e0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.128.0.0/10: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddc440 requirements:0xc000ddc500 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.192.0.0/11: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddc540 requirements:0xc000ddc660 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.224.0.0/12: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddc6c0 requirements:0xc000ddc780 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.240.0.0/13: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddc800 requirements:0xc000ddc8c0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.248.0.0/14: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddc940 requirements:0xc000ddca20 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.252.0.0/15: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddca60 requirements:0xc000ddcb00 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.255.0.0/16: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddcbc0 requirements:0xc000ddcd00 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.0.0/17: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddcd60 requirements:0xc000ddce00 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.192.0/18: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddcfe0 requirements:0xc000ddd1e0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.128.0/19: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddd440 requirements:0xc000ddd580 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.176.0/20: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000ddd640 requirements:0xc000ddd940 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.160.0/21: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dddda0 requirements:0xc000dc4060 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.172.0/22: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc40a0 requirements:0xc000dc41a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.170.0/23: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc4200 requirements:0xc000dc42e0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.168.0/24: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc43a0 requirements:0xc000dc4440 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.0/25: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc4480 requirements:0xc000dc4580 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.128/26: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc45c0 requirements:0xc000dc46a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.192/27: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc4700 requirements:0xc000dc4780 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.224/28: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc47e0 requirements:0xc000dc4880 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.240/29: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc4900 requirements:0xc000dc49a0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.248/30: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc4a80 requirements:0xc000dc4b40 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.252/31: ,},MatchExpressions:[]LabelSelectorRequirement{},}} {LabelSelector:0xc000dc4ba0 requirements:0xc000dc4ca0 cachedLabelSelectorString:&LabelSelector{MatchLabels:map[string]string{cidr.169.254.169.255/32: ,},MatchExpressions:[]LabelSelectorRequirement{},}}]}] Labels:[k8s:io.cilium.k8s.policy.derived-from=NetworkPolicy k8s:io.cilium.k8s.policy.name=singleuser k8s:io.cilium.k8s.policy.namespace=recommender k8s:io.cilium.k8s.policy.uid=e82adfaf-34ab-11eb-9d7a-0296c41eb8a2] Description:}]" policyAddRequest=e82bcde7-34ab-11eb-a8dc-067e3082833c subsys=daemon
level=info msg="Setting up base BPF datapath (BPF v1 instruction set, ktime clock source)" subsys=datapath-loader
level=info msg="Setting sysctl net.core.bpf_jit_enable=1" subsys=datapath-loader
level=info msg="Setting sysctl net.ipv4.conf.all.rp_filter=0" subsys=datapath-loader
level=info msg="Setting sysctl kernel.unprivileged_bpf_disabled=1" subsys=datapath-loader
level=info msg="Blacklisting local route as no-alloc" route=10.0.2.0/24 subsys=ipam
level=info msg="Blacklisting local route as no-alloc" route=172.17.0.0/16 subsys=ipam
```

When the issue is propagated to other agents and ALL workers fails like:
```➜ k get po
NAME                                                               READY   STATUS             RESTARTS   AGE
cilium-4k99p                                                       0/1     CrashLoopBackOff   12         8d
cilium-4xcxj                                                       0/1     CrashLoopBackOff   12         20d
cilium-5czz5                                                       0/1     CrashLoopBackOff   13         8d
cilium-5r6fb                                                       0/1     CrashLoopBackOff   12         21d
cilium-8dnlh                                                       1/1     Running            0          53m
cilium-cmskx                                                       0/1     Error              12         8d
cilium-dnws7                                                       0/1     CrashLoopBackOff   11         7d12h
cilium-hd5bh                                                       1/1     Running            0          21m
cilium-hj2dg                                                       0/1     CrashLoopBackOff   13         7d15h
cilium-j6hz8                                                       0/1     CrashLoopBackOff   12         20d
cilium-jn56g                                                       0/1     CrashLoopBackOff   13         8d
cilium-jztv4                                                       0/1     CrashLoopBackOff   10         8d
cilium-lv7xg                                                       0/1     CrashLoopBackOff   13         8d
cilium-n9wqc                                                       0/1     CrashLoopBackOff   11         3h6m
cilium-operator-695dff44df-jd2bw                                   1/1     Running            1          20d
cilium-pcxp2                                                       1/1     Running            0          42m
cilium-qvqmj                                                       0/1     CrashLoopBackOff   13         21d
cilium-qvrzt                                                       0/1     CrashLoopBackOff   10         20d
cilium-sgbsv                                                       0/1     CrashLoopBackOff   13         59m
cilium-ssn9s                                                       0/1     CrashLoopBackOff   13         8d
cilium-tgrhl                                                       0/1     CrashLoopBackOff   12         20d
cilium-tpg2p                                                       0/1     CrashLoopBackOff   12         21d
cilium-vmvqb                                                       0/1     CrashLoopBackOff   12         21d
cilium-xsts6                                                       0/1     CrashLoopBackOff   12         8d
cluster-autoscaler-aws-cluster-autoscaler-chart-c79cdc86d-bx2w5    0/1     CrashLoopBackOff   12         57m
coredns-5c7dc66fdd-28rz2                                           0/1     Running            11         20d
coredns-5c7dc66fdd-5pxw6                                           0/1     Running            10         20d
coredns-autoscaler-796d6875db-mg4h7                                1/1     Running            0          20d
```

Then, delete helm chart and delete ALL agents to be recreated worked.

The thing that one application in a particular state can break the entire cluster scares me a lot so I still investigating and feel free to request more information

--- a/daemon/cmd/daemon.go
+++ b/daemon/cmd/daemon.go
@@ -625,6 +625,7 @@ func NewDaemon(ctx context.Context, epMgr *endpointmanager.EndpointManager, dp d
 		return nil, nil, err
 	}
 
+	// Must occur after d.allocateIPs(), see GH-14245 and its fix.
 	d.nodeDiscovery.StartDiscovery(nodeTypes.GetName())
 
 	// Annotation of the k8s node must happen after discovery of the

--- a/daemon/cmd/datapath.go
+++ b/daemon/cmd/datapath.go
@@ -56,6 +56,7 @@ import (
 
 // LocalConfig returns the local configuration of the daemon's nodediscovery.
 func (d *Daemon) LocalConfig() *datapath.LocalNodeConfiguration {
+	<-d.nodeDiscovery.Registered
 	return &d.nodeDiscovery.LocalConfig
 }
 

